morhendos/dontsign
# Add dark mode support

This PR adds dark mode support to the application. Changes include:

1. Added ThemeProvider from next-themes
2. Created ThemeToggle component with light/dark/system options
3. Updated root layout to support themes
4. Added theme toggle button to main page

The implementation uses next-themes library which is compatible with Next.js 14 and works well with Tailwind CSS. The theme toggle is positioned in the top-right corner and includes a smooth transition animation.

Testing Instructions:
1. Click the sun/moon icon in the top-right corner
2. Try switching between light, dark, and system themes
3. Verify that all components render correctly in both themes
4. Check that the theme preference persists after page refresh

Note: This PR is based on the `feature/analysis-progress-bar` branch and includes all its changes.

Dependencies to install:
```bash
npm install next-themes
```
